json-transformer-jolt/src/main/java/uk/gov/justice/json/jolt/JsonAgainstSchemaValidator.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.json.jolt;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.io.Resources;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class JsonAgainstSchemaValidator {
+
+    public List<String> validateAgainstSchema(final String schemaFileName,
+                                              final String jsonString) throws TransformationException {
+
+        try (InputStream inputStream = Resources.getResource(schemaFileName).openStream()) {
+
+            final JSONObject rawSchema = new JSONObject(new JSONTokener(inputStream));
+            final Schema schema = SchemaLoader.load(rawSchema);
+            schema.validate(new JSONObject(jsonString));
+
+        } catch (final ValidationException e) {
+            final ArrayList<String> errorMessages = new ArrayList<>();
+
+            if (e.getCausingExceptions().size()==0){
+                errorMessages.add(e.getMessage());
+            }
+            return causingExceptionInfo(e.getCausingExceptions(), errorMessages);
+        } catch (final IOException | IllegalArgumentException e) {
+            throw new TransformationException("Error validating payload against schema, File not readable or available", e);
+        }
+        return new ArrayList<>();
```

Comment:
use Collections.emptyList() would be better...Collections can be static import

Reply:
Done